op-devstack/presets/interop.go

```diff
@@ -84,44 +133,17 @@ func WithUnscheduledInterop() stack.CommonOption {
 }
 
 func NewSimpleInterop(t devtest.T) *SimpleInterop {
-	system := shim.NewSystem(t)
+	singleChain := NewSingleChainInterop(t)
 	orch := Orchestrator()
-	orch.Hydrate(system)
-
-	t.Gate().GreaterOrEqual(len(system.Supervisors()), 1, "expected at least one supervisor")
-	// At this point, any supervisor is acceptable but as the DSL gets fleshed out this should be selecting supervisors
-	// that fit with specific networks and nodes. That will likely require expanding the metadata exposed by the system
-	// since currently there's no way to tell which nodes are using which supervisor.
-
-	t.Gate().Equal(len(system.TestSequencers()), 1, "expected exactly one test sequencer")
-
-	l1Net := system.L1Network(match.FirstL1Network)
-	l2A := system.L2Network(match.Assume(t, match.L2ChainA))
-	l2B := system.L2Network(match.Assume(t, match.L2ChainB))
+	l2B := singleChain.system.L2Network(match.Assume(t, match.L2ChainB))
 	out := &SimpleInterop{
-		Log:           t.Logger(),
-		T:             t,
-		system:        system,
-		TestSequencer: dsl.NewTestSequencer(system.TestSequencer(match.Assume(t, match.FirstTestSequencer))),
-		Supervisor:    dsl.NewSupervisor(system.Supervisor(match.Assume(t, match.FirstSupervisor)), orch.ControlPlane()),
-		ControlPlane:  orch.ControlPlane(),
-		L1Network:     dsl.NewL1Network(l1Net),
-		L1EL:          dsl.NewL1ELNode(l1Net.L1ELNode(match.Assume(t, match.FirstL1EL))),
-		L2ChainA:      dsl.NewL2Network(l2A),
-		L2ChainB:      dsl.NewL2Network(l2B),
-		L2ELA:         dsl.NewL2ELNode(l2A.L2ELNode(match.Assume(t, match.FirstL2EL))),
-		L2ELB:         dsl.NewL2ELNode(l2B.L2ELNode(match.Assume(t, match.FirstL2EL))),
-		L2CLA:         dsl.NewL2CLNode(l2A.L2CLNode(match.Assume(t, match.FirstL2CL)), orch.ControlPlane()),
-		L2CLB:         dsl.NewL2CLNode(l2B.L2CLNode(match.Assume(t, match.FirstL2CL)), orch.ControlPlane()),
-		Wallet:        dsl.NewHDWallet(t, devkeys.TestMnemonic, 30),
-		FaucetA:       dsl.NewFaucet(l2A.Faucet(match.Assume(t, match.FirstFaucet))),
-		FaucetB:       dsl.NewFaucet(l2B.Faucet(match.Assume(t, match.FirstFaucet))),
-		L2BatcherA:    dsl.NewL2Batcher(l2A.L2Batcher(match.Assume(t, match.FirstL2Batcher))),
-		L2BatcherB:    dsl.NewL2Batcher(l2B.L2Batcher(match.Assume(t, match.FirstL2Batcher))),
+		SingleChainInterop: *singleChain,
+		L2ChainB:           dsl.NewL2Network(l2B),
+		L2ELB:              dsl.NewL2ELNode(l2B.L2ELNode(match.Assume(t, match.FirstL2EL))),
+		L2CLB:              dsl.NewL2CLNode(l2B.L2CLNode(match.Assume(t, match.FirstL2CL)), orch.ControlPlane()),
+		FaucetB:            dsl.NewFaucet(l2B.Faucet(match.Assume(t, match.FirstFaucet))),
+		L2BatcherB:         dsl.NewL2Batcher(l2B.L2Batcher(match.Assume(t, match.FirstL2Batcher))),
```

Comment:
Why are we modifying the `SimpleInterop` preset in this change?
Wouldn't we want a 'simple interop' preset to have multiple L2's?

Reply:
The "chainA" bundle of presented DSL components is the same as a single-chain bundle of components. To make it two chains, we just add the components for "chainB". So yes, single-chain is equivalent with chainA. But it's just the DSL presentation; the configuration of these components may be different, but that doesn't matter to the DSL here.